rq/job.py

```diff
@@ -147,6 +151,8 @@ def create(
         description: Optional[str] = None,
         depends_on: Optional[JobDependencyType] = None,
         timeout: Optional[int] = None,
+        success_callback_timeout: Optional[int] = None,
```

Comment:
I like the idea of having separate timeouts for success and failure callbacks.

Perhaps we can have a `Callback` class that can be used like this:
```python
queue.enqueue(say_hello, on_success=Callback(do_something, timeout=30))
```